samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/SubmitToGradingActionListener.java

```diff
@@ -707,28 +707,40 @@ private void prepareItemGradingPerItem(ActionEvent ae, DeliveryBean delivery,
 		case 15: // CALCULATED_QUESTION
 		case 16: //IMAGEMAP_QUESTION 	
 		case 11: // FIN
-			boolean addedToAdds = false;
 			for (int m = 0; m < grading.size(); m++) {
 				ItemGradingData itemgrading = grading.get(m);
 				itemgrading.setAgentId(AgentFacade.getAgentString());
 				itemgrading.setSubmittedDate(new Date());
 			}
+			int fake=-1;
 			for (int m = 0; m < grading.size(); m++) {
 				ItemGradingData itemgrading = grading.get(m);
+				String s = itemgrading.getAnswerText();
 				if (itemgrading.getItemGradingId() != null
 						&& itemgrading.getItemGradingId().intValue() > 0) {
-					adds.addAll(grading);
-					break;
-				} else if (itemgrading.getAnswerText() != null && !itemgrading.getAnswerText().equals("")) {
-					String s = itemgrading.getAnswerText();
+					 if ("1".equals(delivery.getNavigation()) && itemgrading.getPublishedAnswerId()==null && (s==null || s.equals(""))) {
+						//Mark this as the fake itemgrading record
+						{fake=m;}	 
+				     } else {
+						 log.debug("s = " + s);
+						 // Change to allow student submissions in rich-text [SAK-17021]
+						 itemgrading.setAnswerText(s);
+						 adds.add(itemgrading);
+				     }	
+				}
+				else if (s != null && !s.equals("")) {
```

Comment:
If StringUtils is available here, it's preferable

Reply:
changed and use in 2 placed in the modified code

Reply:
thanks